src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs

```diff
@@ -36,11 +36,30 @@ internal static class WebResponseHelper
         }
 
         internal static string GetOutFilePath(HttpResponseMessage response, string _qualifiedOutFile)
-        {
-            // Get file name from last segment of Uri
-            string lastUriSegment = System.Net.WebUtility.UrlDecode(response.RequestMessage.RequestUri.Segments[^1]);
+        {            
+            if (!Directory.Exists(_qualifiedOutFile))
+            {
+                return _qualifiedOutFile;
+            }
+
+            string contentDisposition = response.Content.Headers.ContentDisposition?.FileNameStar ?? response.Content.Headers.ContentDisposition?.FileName;
+
+            if (!string.IsNullOrEmpty(contentDisposition)) 
+            {
+                // Get file name from Content-Disposition header if present
+                return Path.Join(_qualifiedOutFile, contentDisposition);
+            }
+
+            if (response.RequestMessage.RequestUri.PathAndQuery != "/")
```

Comment:
Could you please clarify why we need PathAndQuery. Maybe add a comment in the code.

Reply:
We need it to handle this case:
```ps1
Invoke-WebRequest https://www.google.com/ -OutFile .\  #PathAndQuery == "/"
#--> OutFile: www_google_com
```